modules/oe_subscriptions_anonymous/src/Controller/SubscriptionAnonymousController.php

```diff
@@ -96,9 +97,25 @@ public function confirmSubscriptionRequest(FlagInterface $flag, string $entity_i
       return $response;
     }
 
-    $this->subscriptionManager->subscribe($email, $flag, $entity_id);
     // The token has been used, so we need to invalidate it.
     $this->tokenManager->delete($email, $scope);
+
+    try {
+      $this->subscriptionManager->subscribe($email, $flag, $entity_id);
+    }
+    catch (RegisteredUserEmailException) {
```

Comment:
We could rethink the return behavior of the `->subscribe()` method.
Instead of an exception, there could be a return value indicating different types of success or failure.
For the first version I wanted to avoid scope creep.

Reply:
Also note that the proposed code change omits the `$e` variable.
The reason is that the exception message or stack does not tell us anything relevant.
We know there is exactly one reason why the exception would be thrown.